vyper/utils.py

```diff
@@ -158,43 +155,49 @@ def in_bounds(cls, type_str, value):
     'constant',
 } | valid_units | valid_call_keywords
 
+# Available base types
+base_types = {'int128', 'decimal', 'bytes32', 'uint256', 'bool', 'address'}
 
 # Cannot be used for variable or member naming
 reserved_words = {
-    # types
-    'int128', 'uint256',
-    'address',
-    'bytes32',
-    'map',
-    'string', 'bytes',
+    # reference types
+    'map', 'string', 'bytes',
+    # decorators
+    'public', 'constant', 'private', 'payable', 'nonreentrant',
```

Comment:
still feels a bit weird because these are also in `valid_global_keywords`, but maybe it's not too important

Reply:
Yeah, I mean it removes duplicates. But I could move decorators up to that.

Reply:
well, i see why it's done the way it is. I think this way is clear enough, no changes needed.